registry/storage/catalog.go

```diff
@@ -95,3 +95,23 @@ func (reg *registry) Enumerate(ctx context.Context, ingester func(string) error)
 	return nil
 
 }
+
+func pathGreaterThan(pathX, pathY string) (b bool) {
+	splitPathX := strings.SplitN(pathX, "/", 2)
+	splitPathY := strings.SplitN(pathY, "/", 2)
+
+	if splitPathX[0] == splitPathY[0] {
+		if len(splitPathX) == 1 && len(splitPathY) == 1 {
+			return false
+		} else if len(splitPathX) == 1 && len(splitPathY) != 1 {
+			return false
+		} else if len(splitPathX) != 1 && len(splitPathY) == 1 {
+			return true
+		}
+
+		return pathGreaterThan(splitPathX[1], splitPathY[1])
```

Comment:
The test will  fail because we expect the registry to have 9 entries . If you add the two entries    "foo-bar/baz-bat/a" and     "foo-bar/baz-bat/b" the registry wont return EOF because there are 2 more entries.

The error I have when add the two entries (in repos and expectd [] string) is 
```
	catalog_test.go:160: Expected end of catalog
	catalog_test.go:175: Expected catalog fourth chunk err
```

Which confirm my previous statement.